src/logic/currentSession/store/reducer/currentSession.ts

```diff
@@ -7,24 +6,23 @@ import { saveCurrentSessionToStorage } from 'src/logic/currentSession/utils'
 
 export const CURRENT_SESSION_REDUCER_ID = 'currentSession'
 
-export type SerializedSessionState = {
-  viewedSafes: string[]
-}
+export type SerializedSessionState = string[]
 
 export default handleActions(
   {
-    [LOAD_CURRENT_SESSION]: (state, action) => state.merge(Map(action.payload)),
+    [LOAD_CURRENT_SESSION]: (state, action) => {
+      return action.payload
+    },
     [UPDATE_VIEWED_SAFES]: (state, action) => {
       const safeAddress = action.payload
 
-      const newState = state.updateIn(['viewedSafes'], (prev) =>
-        prev.includes(safeAddress) ? prev : [...prev, safeAddress],
-      )
-
-      saveCurrentSessionToStorage(newState)
+      if (!state.includes(safeAddress)) {
+        state.push(safeAddress)
+        saveCurrentSessionToStorage(state)
+      }
 
-      return newState
+      return state
     },
   },
-  Map(),
+  [],
```

Comment:
So the whole current session is now just an array with addresses with visited safes? What happens if we want to add a new prop there?

Reply:
We are not using that state, I would suggest removing it completely

Reply:
> We are not using that state

It's used in incoming transactions middleware

> What happens if we want to add a new prop there?

This does not really answer the question

Reply:
If you wanna add a prop in the future, just do the change..

Reply:
And refactor other stuff that was dependent on the previous structure? It had a structure where you could easily add new properties and you're changing it. Your change doesn't make any sense and makes it harder for other developers to do any changes in the future. I agree that it would make sense if we wanted to ship the feature really hard but your change is making something bad that was done in a good way